packages/create-schemas/src/plugins/openapi-fetch-plugin.ts

```diff
@@ -0,0 +1,25 @@
+import type { Plugin } from "./plugin.ts";
+import { getRelativeModuleResolutionExtension } from "../utils.ts";
+import { openapiTypeScriptId } from "./openapi-typescript-plugin.ts";
+
+export function openapiFetchPlugin(): Plugin {
+    return {
+        name: "openapi-fetch-plugin",
+        async transform({ id, emitFile }) {
+            if (id !== openapiTypeScriptId) {
+                return;
+            }
+
+            const importsFileExtension = getRelativeModuleResolutionExtension();
+
+            emitFile({
+                filename: "client.ts",
+                code: [
+                    `import type { paths } from "./types${importsFileExtension}";`,
+                    "import _createClient from \"openapi-fetch\";",
+                    "export const createClient = _createClient as typeof _createClient<paths, \"application/json\">;"
```

Comment:
just to confirm, are we always expecting the endpoint return media type to be `application/json`? (which is fine and is per tech guidelines, this is just for my understanding)

Reply:
Yep